### PR TITLE
feat: added new select menu types & new field component::channel_types

### PR DESCRIPTION
**Pull request by Commandserver**
fix: #528

### CODE REVIEW

File: src/dpp/message.cpp [hunk 1]
```diff
@@ -268,9 +288,8 @@ void to_json(json& j, const component& cp) {
 			j["emoji"]["name"] = cp.emoji.name;
 		}
 	} else if (cp.type == cot_selectmenu) {
-		j["type"] = cp.type;
 		j["custom_id"] = cp.custom_id;
-		//j["disabled"] = cp.disabled;
+		j["disabled"] = cp.disabled;
```
[the remark this answers is not in the document]
Comment (braindigitalis) [reply src/dpp/message.cpp:292]: it was not supported here originally, there was an issue too where if you defined it at all it was disabled so not setting it was not disabled and any value was disabled.

if you've tested it and it's no longer the case lets put this fix in 😉

File: include/dpp/message.h [hunk 2]
```diff
@@ -40,10 +40,18 @@ enum component_type : uint8_t {
 	cot_action_row = 1,
 	/// Clickable button
 	cot_button = 2,
-	/// Select menu
+	/// Select menu for picking from defined text options
 	cot_selectmenu = 3,
```
[the remark this answers is not in the document]
Comment (braindigitalis) [reply include/dpp/message.h:44]: I don't think it needs renaming, that would be a breaking change

File: src/dpp/message.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Commandserver) [reply src/dpp/message.cpp:292]: i've tested it with message_create. had no problems with that 🤷

**Comment by Commandserver** [comment]
where can i add the word to spellcheck?

**Comment by Henny022** [comment]
I tried using this, but it seems to be lacking proper handling of the select event.
I tried duplicating the responsible code in `interaction_create::handle` but for some reason it couldn't properly parse the `values` field, I couldn't figure out why

**Comment by Commandserver** [comment]
> I tried using this, but it seems to be lacking proper handling of the select event.
> I tried duplicating the responsible code in `interaction_create::handle` but for some reason it couldn't properly parse the `values` field, I couldn't figure out why

What handling do you mean @Henny022 ?

**Comment by braindigitalis** [comment]
> where can i add the word to spellcheck?

add the word in the `.cspell.json` at repository root

**Comment by Henny022** [comment]
I mean the handler for the discord event when using ie. a channel select.
The text select has `on_select_click`, but it does not work for the others as is because the `component_type` is different

**Comment by Commandserver** [comment]
> I mean the handler for the discord event when using ie. a channel select. The text select has `on_select_click`, but it does not work for the others as is because the `component_type` is different

Thank you so much :D, should be fixed.

File: include/dpp/appcommand.h [hunk 3]
```diff
@@ -601,26 +601,12 @@ struct DPP_EXPORT command_interaction {
...
-	/**
-	 * @brief Button
-	 */
-	cotype_button = 2,
-	/**
-	 * @brief Option select list (drop-down)
-	 */
-	cotype_select = 3
-};
-
```
Comment (Commandserver) [inline include/dpp/appcommand.h:617]: `dpp::component_type` is now used instead because it was just a duplicate of it.

**Comment by Commandserver** [comment]
I've tested channel and role select menus.
The only issue i've ran in, is that the IDs for e.g. channels are stored as strings in the `select_click_t::values` array. Would be nice if an additional field could be added for the selected ids or so

**Pull request merged**